river_dl/RGCN.py

```diff
@@ -108,25 +123,45 @@ def __init__(self, hidden_size, A, flow_in_temp=False, rand_seed=None):
                 shape=[1], initializer="zeros", name="b_out"
             )
         else:
-            # was W2
-            self.W_out = self.add_weight(
-                shape=[hidden_size, 2], initializer=w_initializer, name="W_out"
-            )
-            # was b2
-            self.b_out = self.add_weight(
-                shape=[2], initializer="zeros", name="b_out"
-            )
+            if self.tasks == 2: 
+                # was W2
+                self.W_out = self.add_weight(
+                    shape=[hidden_size, 2], initializer=w_initializer, name="W_out"
+                )
+                # was b2
+                self.b_out = self.add_weight(
+                    shape=[2], initializer="zeros", name="b_out"
+                )
+            else: 
+                # was W2
+                self.W_out = self.add_weight(
+                    shape=[hidden_size, 1], initializer=w_initializer, name="W_out"
+                )
+                # was b2
+                self.b_out = self.add_weight(
+                    shape=[1], initializer="zeros", name="b_out"
+                )
 
     @tf.function
     def call(self, inputs, **kwargs):
+        h_list = []
+        c_list = []
         graph_size = self.A.shape[0]
         hidden_state_prev, cell_state_prev = (
             tf.zeros([graph_size, self.hidden_size]),
             tf.zeros([graph_size, self.hidden_size]),
         )
         out = []
         n_steps = inputs.shape[1]
+        h_update = tf.cast(kwargs['h_init'], tf.float32)
+        c_update = tf.cast(kwargs['c_init'], tf.float32)
+        if self.return_state:
+            # set the initial h & c states to the supplied h and c states if using DA 
+            hidden_state_prev = h_update 
+            cell_state_prev = c_update 
```

Comment:
Is it necessary to define `h_update` and `c_update` here rather than waiting until line 184? They don't seem to get used before being overwritten, except in line 163 which itself doesn't seem to get used. (Or does `self.lstm` change the properties of `self`, such that line 163 does something even though `seq` and `state` go unused?) I'm wondering if this block of 12 lines could be simplified:
```suggestion
        out = []
        n_steps = inputs.shape[1]
        # set the initial h & c states to the supplied h and c states if using DA, or 0's otherwise
        if self.return_state:
            hidden_state_prev = tf.cast(kwargs['h_init'], tf.float32)
            cell_state_prev = tf.cast(kwargs['c_init'], tf.float32)
        else:
            hidden_state_prev = tf.zeros([graph_size, self.hidden_size])
            cell_state_prev = tf.zeros([graph_size, self.hidden_size])
```

Reply:
good call, I think that still works and is much cleaner. I'm pretty sure the call on 163 was leftover from previous versions of the river-dl code as I had those lines in an earlier version.